projects/plugins/wpcomsh/custom-colors/colors.php

```diff
@@ -1483,17 +1483,33 @@ public static function print_theme_css() {
 	}
 
 	/**
-	 * Print block editor CSS.
+	 * Adds block editor CSS to the block editor settings.
+	 *
+	 * @param array $editor_settings Block editor settings.
 	 */
-	public static function print_block_editor_css() {
+	public static function add_block_editor_css( $editor_settings ) {
 		if ( ! self::should_enable_colors() ) {
-			return;
+			return $editor_settings;
 		}
+
 		$css = self::get_theme_css();
 
-		wp_register_style( 'custom-colors-editor-css', false, array(), '20210311' ); // Register an empty stylesheet to append custom CSS to.
-		wp_enqueue_style( 'custom-colors-editor-css' );
-		wp_add_inline_style( 'custom-colors-editor-css', $css ); // Append inline style to our new stylesheet
+		if ( is_array( $editor_settings['styles'] ) ) {
+			$editor_settings['styles'] [] = array(
+				'css'            => $css,
+				'__unstableType' => 'theme',
+				'isGlobalStyles' => false,
+			);
+		} else {
+			$editor_settings['styles'] = array(
+				array(
+					'css'            => $css,
+					'__unstableType' => 'theme',
+					'isGlobalStyles' => false,
+				),
+			);
+		}
+		return $editor_settings;
```

Comment:
Reading this code, it seems what we want is to "add an editor style". But we want that style to be dynamic (generated in the server). 

Unfortunately `add_editor_style`'s WP function doesn't support adding inline styles (it only sources from files and urls). So I think personally if a ticket or issue doesn't exist on WordPress Core, it's worth creating. (A dedicated function to add an inline editor style).

I know the code here does the same, but for me that's a bit too low level.

Reply:
Yes, I think an add_editor_styles function which receives dynamic styles instead of a file or URL may make sense. I will create an issue.